docs/build/js-client/4-advanced-ipfs.md

```diff
@@ -0,0 +1,297 @@
+            const content = await timeout(s.get(c, { signal: controller.signal }), 5 /* 5 sec */, controller).catch(noop);
+
+            const result = Boolean(content);
+            await helia.stop();
+            return result;
+        },
+        async remove(cid: string) {
+            const helia = await makeHelia();
+
+            const c = CID.parse(cid);
+
+            const isPinned = await helia.pins.isPinned(c); // Pinning protects data from garbage collector in IPFS
+
+            if (isPinned) {
+                console.log('Remove pinned entry:', c.toString());
+                const pin = await helia.pins.rm(c);
+                await helia.gc();
+            }
+
+            await helia.stop();
+            return c.toString();
+        },
+        async upload(path: string) {
+            const helia = await makeHelia();
+
+            const s = strings(helia);
+
+            const buffer = await readFile(path);
+
+            const cid = await s.add(buffer.toString());
+            await helia.pins.add(cid);
+
+            await helia.stop();
+            return cid.toString();
+        },
+        async upload_string(contents: string) {
+            const helia = await makeHelia();
+
+            const s = strings(helia);
+
+            const cid = await s.add(contents);
+            await helia.pins.add(cid);
+
+            await helia.stop();
+            return cid.toString();
+        },
+        async dag_upload(path: string) {
+            const helia = await makeHelia();
+
+            const dj = dagJson(helia);
+
+            const buffer = await readFile(path);
+
+            const cid = await dj.add(buffer.toString());
+            await helia.pins.add(cid);
+
+            await helia.stop();
+            return cid.toString();
+        },
+        async dag_upload_string(contents: string) {
+            const helia = await makeHelia();
+
+            const dj = dagJson(helia);
+
+            const cid = await dj.add(contents);
+            await helia.pins.add(cid);
+
+            await helia.stop();
+            return cid.toString();
+        }
+    });
+
+    const cid = await upload_string('Hello world!!!', { ttl: 120000 });
+
+    console.log('cid:', cid.toString());
+
+    console.log('Is entry exists:', await exists(cid.toString(), { ttl: 120000 }));
+
+    await remove(cid.toString(), { ttl: 120000 });
+
+    console.log('Is entry exists:', await exists(cid.toString(), { ttl: 120000 }));
+
+    await Fluence.disconnect();
+}
+
+main();
+```
+
+Let's try running the example:
+
+```sh
+./node_modules/.bin/ts-node-esm ./src/index.ts
+```
+
+If everything has been done correctly, you should see the following text in the console:
+
+```
+id: bafkreicdktp5u4gi6djzsg454pkw3s3ot4x4nqbrnurvwy5p5m4ii4nnuq
+Is entry exists: true
+Remove pinned entry: bafkreicdktp5u4gi6djzsg454pkw3s3ot4x4nqbrnurvwy5p5m4ii4nnuq
+Is entry exists: false
+```
+
```

Comment:
Need some kind of a recap of what happened, what did reader just learned and what to do next